### ISSUE TITLE
How to chain animations?

### ISSUE THREAD

**Issue by mori-honest** [daimajia/AndroidViewAnimations#67]

**Comment by HarshavardhanHokage** [comment]
Right now I don't think repeat is supported but you can try the following work around. Create a method. Drop your animation within it and then add a listener to it. In OnAnimationEnd call the method again.

Example:
'''
 public void doYoyo(final Techniques techniques,final View view)
    {
        YoYo.with(techniques).duration(2000).interpolate(new LinearInterpolator()).withListener(new Animator.AnimatorListener() {
            @Override
            public void onAnimationStart(Animator animation) {

            }

            @Override
            public void onAnimationEnd(Animator animation) {
                doYoyo(techniques, view);
            }

            @Override
            public void onAnimationCancel(Animator animation) {

            }

            @Override
            public void onAnimationRepeat(Animator animation) {

            }
        }).playOn(view);
    }
'''

**Comment by kashban** [comment]
You could use Listeners:

    YoYo.with(Techniques.FadeIn).duration(250)
                .withListener(new Animator.AnimatorListener() {
                    @Override
                    public void onAnimationStart(Animator animation) {
                       
                    }

                    @Override
                    public void onAnimationEnd(Animator animation) {
                         // Start new animation here
                    }

                    @Override
                    public void onAnimationCancel(Animator animation) {}

                    @Override
                    public void onAnimationRepeat(Animator animation) {
                           // This is being called when the animation starts over, I guess
                    }
                })
                .playOn(lyt_ProgressBarOverlay);

**Comment by mori-honest** [comment]
I know this, but it would be great to have a set of animations played on a single view one after another or together.

**Comment by kashban** [comment]
@mori-honest I agree, something like a choregrapher where you put together multiple animations, be it chained or in parallel would be nice.

**Comment by aliarasteh** [comment]
@mori-honest Exactly using set of animations would be much usable rather than applying a single animation effect.

**Comment by saglamhkn** [comment]
private void animateButtonsChain(final View[] views, final Techniques technique, final boolean isShow) {
        animate(views[0], technique, isShow, new OnClickListener() {
            int counter = 1;
            @Override
            public void onClick(View view) {
                if (counter < views.length )
                    animate(views[counter++], technique, isShow, this);
            }
        });
    }

    private void animate(final View v, Techniques t, final boolean isShow, final OnClickListener l) {
        YoYo.with(t).withListener(new Animator.AnimatorListener() {
            @Override
            public void onAnimationStart(Animator animator) {
                if (isShow)
                    v.setVisibility(VISIBLE);
            }

            @Override
            public void onAnimationEnd(Animator animator) {
                if (!isShow)
                    v.setVisibility(GONE);
                l.onClick(v);
            }

            @Override
            public void onAnimationCancel(Animator animator) {

            }

            @Override
            public void onAnimationRepeat(Animator animator) {

            }
        }).repeat(0).duration(150).playOn(v);
    }

**Comment by entrealist** [comment]
Kotlin version of @saglamhkn code. Working correctly :+1: 
```
private fun animateButtonsChain(
        views: ArrayList<View>,
        technique: Techniques,
        visible: Boolean,
    ) {
        animate(views[0], technique, visible, object : View.OnClickListener {
            var counter = 1

            override fun onClick(view: View) {
                if (counter < views.size)
                    animate(views[counter++], technique, visible, this)
            }
        })
    }

    private fun animate(
        view: View,
        technique: Techniques,
        visible: Boolean,
        clickListener: View.OnClickListener,
    ) {
        YoYo.with(technique)
            .duration(ANIMATION_DURATION)
            .repeat(0)
            .withListener(object : Animator.AnimatorListener {
                override fun onAnimationStart(animator: Animator) {
                    view.visibility = if (visible) VISIBLE else INVISIBLE
                }

                override fun onAnimationEnd(animator: Animator) {
                    view.visibility = if (!visible) INVISIBLE else VISIBLE 
                    clickListener.onClick(view)
                }

                override fun onAnimationCancel(animator: Animator) {
                }

                override fun onAnimationRepeat(animator: Animator) {
                }
            })
            .playOn(view)
    }
```
